src/Filters/AbstractAuthFilter.php

```diff
@@ -30,6 +30,12 @@ public function before(RequestInterface $request, $arguments = null)
         }
 
         if (! auth()->loggedIn()) {
+            // Set the entrance url to redirect a user after successful login
+            if (! url_is('login')) {
+                $session = session();
+                $session->setTempdata('beforeLoginUrl', current_url(), 300);
```

Comment:
I just came up with this code, but it is a copy and paste. 
Since it is the same knowledge (save the URL for 300 seconds in the session), it might be better to put it together in a method if possible.

But there doesn't seem to be an appropriate class to put it in now.